### PR TITLE
init

### PR DESCRIPTION
**Pull request by djia-vm-wf**

### CODE REVIEW

**Review by akodali18 — COMMENTED** [review]
Can we add unit tests for this as well.

File: lib/helper.js [hunk 1]
```diff
@@ -24,6 +24,10 @@ function encodeKey(metricName, tags) {
   }
   // Sort the tags based on their keys to ensure we don't duplicate metrics if the same tags come out of order.
   var tagsArray = Object.keys(tags).sort().map(key => [key, tags[key]]);
+  var tagsType = typeof tags;
```
Comment (aliu-vmware) [inline lib/helper.js]: Couple of things here:
1. These checks should be done before tagsArray is executed.
2. typeof and types in general are very tricky in javascript.  You should try to handle as many cases as possible.  Please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Examples for the sort of weirdness that can occur.
3. prefer !== over != in javascript
4. avoid throwing non-Errors.  This is easily fixed by switching to `throw new Error("message here")`

File: lib/helper.js [hunk 2]
```diff
@@ -24,6 +24,10 @@ function encodeKey(metricName, tags) {
...
+  if (tagsType != "object") {
+    throw "Wrong Tags Object Sent To The API"
+  }
   return metricName + tagSeparator + JSON.stringify(tagsArray) ;
```
Comment (aliu-vmware) [inline lib/helper.js]: optional: prefer template string construction over string addition.  See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: test/helper.js [hunk 3]
```diff
@@ -27,3 +28,13 @@ describe('decodeKey', function() {
     expect(decodedKey[1]).to.be.equal(null);
   });
 });
+
+describe('encodeKey', function () {
+  it('wrong tags API', function () {
+    const registry = new metrics.Registry();
+    let c = new metrics.Counter();
+    expect( function () {
+      registry.addTaggedMetric("request.counter1", c, "{\"key1\":\"val1\"}");
```
Comment (aliu-vmware) [inline test/helper.js]: should have multiple cases trying other weird stuff: null, Date, array, etc.

**Review by aliu-vmware — CHANGES_REQUESTED** [review]
See inline comments.

File: lib/helper.js [hunk 4]
```diff
@@ -24,6 +24,10 @@ function encodeKey(metricName, tags) {
   }
   // Sort the tags based on their keys to ensure we don't duplicate metrics if the same tags come out of order.
   var tagsArray = Object.keys(tags).sort().map(key => [key, tags[key]]);
+  var tagsType = typeof tags;
+  if (tagsType != "object") {
+    throw "Wrong Tags Object Sent To The API"
```
Comment (akodali18) [inline lib/helper.js]: We can make this exception message as clear to the user as possible with info about "why it failed" and "the expected format"

**Review by aliu-vmware — APPROVED** [review]

**Review by akodali18 — COMMENTED** [review]
The second change - https://github.com/wavefrontHQ/nodejs-metrics-wavefront/pull/20/commits/b7b15c1ea1eb695b1e3467672a928163aa892587 seems to have broken the sdk build. 
Rest LGTM.

**Review by akodali18 — APPROVED** [review]

**Pull request merged**